joomla/joomla-cms
# [4.0] Client performance improvement for plg_system_debug

A constant issue for me is the performance when enabling the debug mode on pages with lots of queries. At some point, the display of the Database Queries tab brings my browser to a grinding halt and opening one of the accordions then fails completely. As a specific example: I did some finder tests and that generated 200 queries. Clicking on an Explain or Profile accordion resulted just in a warning that some JS was slowing down the webpage, but no action on the page itself. Running on FF61 and a Core i7 here...

This PR replaces the bootstrap accordions for each query with a pure CSS implementation, which is much more performant. According to caniuse.com, all the features used in the CSS are IE11 compatible and work in all major browsers. The only browser that supposedly has problems is Opera Browser Mini. The debug.min.css was generated by running ```node build.js --compilecss```.

### How to test
1. Enable the Smart Search content plugin
2. Edit /administrator/components/com_content/Controller/ArticleController.php and insert this method in the class:
	```
		public function save($key = null, $urlVar = null)
		{
			parent::save($key, $urlVar);
			$this->setRedirect(false);
		}
	```
	This prevents Joomla from redirecting after saving an article.
3. Enable debug mode.
4. Edit an article and click on save.
5. See the database queries tab of the debug informations and notice about 200 queries being logged. Try to open an accordion of the queries and notice that it doesn't perform well.
6. Apply PR, reload that page and see that it reacts a lot quicker.
